x-pack/platform/test/security_api_integration/tests/login_selector/basic_functionality.ts

```diff
@@ -575,77 +609,197 @@ export default function ({ getService }: FtrProviderContext) {
-          const callbackFunc = () => {
-            return supertest
-              .post('/api/security/saml/callback')
-              .ca(CA_CERT)
-              .set('Cookie', samlValues.cookie.cookieString())
-              .send({
-                SAMLResponse: samlValues.samlResponse,
-              });
-          };
+          if (firstResponse) responses.unshift(firstResponse);
 
-          preparedCallbacks.push(callbackFunc);
-        }
+          const locations: string[] = [];
+          const statusCodes: number[] = [];
 
-        const responses = await Promise.all(
-          preparedCallbacks.map((func) => {
-            return func();
-          })
-        );
-        expect(
           responses.map((response: Response) => {
-            return response.headers.location;
-          })
-        ).to.eql([
-          '/abc/xyz/0',
-          '/abc/xyz/1',
-          '/abc/xyz/2',
-          '/abc/xyz/3',
-          '/abc/xyz/4',
-          '/abc/xyz/5',
-          '/abc/xyz/6',
-          '/abc/xyz/7',
-          '/abc/xyz/8',
-          '/abc/xyz/9',
-        ]);
+            locations.push(response.headers.location);
+            statusCodes.push(response.statusCode);
+          });
+
+          // Sort locations to guarantee order
+          locations.sort();
+
+          // all URLs should have been redirected to the correct locations
+          expect(locations).to.eql([
+            '/abc/xyz/0',
+            '/abc/xyz/1',
+            '/abc/xyz/2',
+            '/abc/xyz/3',
+            '/abc/xyz/4',
+            '/abc/xyz/5',
+            '/abc/xyz/6',
+            '/abc/xyz/7',
+            '/abc/xyz/8',
+            '/abc/xyz/9',
+          ]);
+
+          // all status codes should be 302
+          expect(statusCodes).to.eql([302, 302, 302, 302, 302, 302, 302, 302, 302, 302]);
+
+          await es.indices.refresh({ index: '.kibana_security_session*' });
+          const finalSessionResponse = await es.search<SessionValue>({
+            index: '.kibana_security_session*',
+            size: 100,
+            query: {
+              bool: {
+                must: [
+                  {
+                    match_all: {},
+                  },
+                ],
+              },
+            },
+          });
+
+          const finalSources = finalSessionResponse.hits.hits.map((hit) => {
+            return {
+              id: hit._id,
+            };
+          });
+
+          // There should be:
+          // - 1 authenticated session that 5 of the callbacks used from the first callback that was "successful"
+          // - 5 authenticated sessions created by the other 5 callbacks that used intermediate cookies
+          // - 1 intermediate session
+          expect(finalSources).to.have.length(7);
+
+          const intermediateSession = finalSources.filter((session) => {
+            return session.id === intermediateSessionId;
+          })[0];
+
+          expect(intermediateSession).to.not.be(undefined);
```

Comment:
__nit:__ can we use `find()` instead of `filter()[0]`?

Reply:
Fixed in https://github.com/elastic/kibana/pull/220426/commits/413cced82b12c201040f71876028729606ebd92c